src/orientationUtils/RelativeDataSupplier.java

```diff
@@ -0,0 +1,74 @@
+package orientationUtils;
+
+import java.util.function.Supplier;
+
+import kalmanFilter.RobotOrientation;
+
+/**
+ * This class represents an object with a changing value. </br>
+ * The object may be a sensor measuring changing values, time-steps, or any
+ * other data supplier.</br>
+ * </br>
+ * 
+ * The instances of this class have one method called {@link #get()}, which
+ * returns the difference between the current measured value and the last
+ * measured value.</br>
+ * </br>
+ * 
+ * Use an instance of this class when you are interested in the difference
+ * between two measurements, and not in each measurment's singular value.
+ * 
+ * <p>
+ * note - in this project, this class is used in order to give us the difference
+ * between two measurements of an encoder instead of the value of each
+ * measurement, and the time passed between each
+ * {@link RobotOrientation#update(measurement,controlChanges, yawAngle,rollAngle, pitchAngle)}
+ * cycle.
+ * </p>
+ * 
+ * @see Supplier
+ * 
+ */
+public class RelativeDataSupplier implements Supplier<Double> {
+	/**
+	 * The {@link Supplier} providing the measured values.
+	 */
+	private Supplier<Double> valueSupplier;
+	/**
```

Comment:
you cant fuckin contain a supplier in a class implementing a supplier #twopoints, fix dis plz

Reply:
we talked about this